front/lib/models.ts

```diff
@@ -419,6 +435,57 @@ DataSource.init(
 );
 User.hasMany(DataSource);
 
+export class Connector extends Model<
+  InferAttributes<Connector>,
+  InferCreationAttributes<Connector>
+> {
+  declare id: CreationOptional<number>;
+  declare createdAt: CreationOptional<Date>;
+  declare updatedAt: CreationOptional<Date>;
+
+  declare type: "slack" | "notion" | "google";
+
+  declare nangoConnectionId: string;
+
+  declare dataSourceId: ForeignKey<DataSource["id"]>;
+  declare userId: ForeignKey<User["id"]>;
+}
+
+Connector.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    createdAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    updatedAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    type: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+    nangoConnectionId: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+  },
+  {
+    modelName: "connector",
+    sequelize: front_sequelize,
+  }
+);
+User.hasMany(Connector);
+DataSource.hasOne(Connector);
```

Comment:
There's a world in which a Connector as multiple DataSources (eg of different max_embededint_size) or with different configs. :thinking:

Reply:
Or maybe a dataSource that generates fact instead of direclty embedding stuff